app/models/post.rb

```diff
@@ -27,7 +27,7 @@ def to_s
     title
   end
 
-  def printLimit
+  def print_limit
```

Comment:
I don't see the problem here. Your logic is complicated (I assume this is necessary) and thus your code becomes complicated.